hapi/src/api/sponsor.api.js

```diff
@@ -23,19 +23,35 @@ const editProfile = async (account, profile) => {
   await historyApi.insert(addSponsorTransaction)
   await userApi.setEmail({ account: { _eq: account } }, profile.email)
 
-  if (profile.geolocation)
-    await locationApi.update(account, {
-      name: profile.name,
-      geolocation: {
-        type: 'Point',
-        coordinates: [
-          profile.geolocation.longitude,
-          profile.geolocation.latitude
-        ]
-      },
-      type: LOCATION_TYPES.SPONSOR,
-      info: profile
-    })
+  if (profile.geolocation) {
+    const { location } = await locationApi.verifyExistence(account)
+    location[0]
```

Comment:
the 97% object properties of the second parameter to update and insert are the same values, we can create an object in order to avoid write repeated code:

example:
```
const data = {
  name: profile.name,
  geolocation: {
    type: 'Point',
    coordinates: [
      profile.geolocation.longitude,
      profile.geolocation.latitude
    ]
  },
  type: LOCATION_TYPES.SPONSOR,
  info: profile
}

location[0]
      ? await locationApi.update(account, data)
      : await locationApi.insert({ ...data, account }) // here just add account property using js destructuring
```

Reply:
Sounds really good. Thank you for that